### ISSUE TITLE
New bug in studpermu.test

### ISSUE THREAD

**Issue by rubak** [spatstat/spatstat#115]
New bug uncovered in comment to [my previous reply on SO](https://stackoverflow.com/a/58321531/3341769). Happens in `studpermu.test` if summary statistic is `Kest` and the first point pattern has less than 3000 points and hence defaults to `valu = "iso"` in the `fv` object and a later pattern has `>=3000` points so only border correction is calculated. Example exposing the bug:

```
library(spatstat)
X <- runifpoint(50, nsim = 3)
Y <- runifpoint(3000, nsim = 3)
h <- hyperframe(ppp = c(X, Y), group = rep(1:2, 3))
studpermu.test(h, ppp ~ group)
```

Workaround:

```
studpermu.test(h, ppp ~ group, correction = "border")
```

Not urgent since workaround was posted on SO.

**Comment by rubak** [comment]
The problematic code is here:
https://github.com/spatstat/spatstat/blob/647d4aacb8976d88fd5fe8ab583834af5aaf7d72/R/studpermutest.R#L221-L227

**Comment by baddstats** [comment]
A similar bug once happened in `envelope`. I will apply a similar fix.

**Comment by baddstats** [comment]
It's fixed in `spatstat 1.61-0.026`
